### PR TITLE
Rake to purge broken visualizations without user table associated

### PR DESCRIPTION
**Pull request by ethervoid**
Related #13928 

We have almost 90k visualization without user_table and we need to remove them

### CODE REVIEW

File: lib/tasks/viz_maintenance.rake [hunk 1]
```diff
@@ -19,6 +19,19 @@ namespace :cartodb do
       end
     end
 
+    desc "Purges broken table visualizations"
+    task :delete_inconsistent_tables => :environment do |t, args|
+      collection = Carto::Visualization.joins("left join user_tables ut on visualizations.map_id = ut.map_id").where("visualizations.type = 'table' and ut.id is null")
+
+      collection.each do |viz|
```
[the remark this answers is not in the document]
Comment (javitonino) [reply lib/tasks/viz_maintenance.rake]: +1. `find_each` works well with UUIDs (basically, with anything that is sortable and doesn't have duplicates)

File: lib/tasks/viz_maintenance.rake [hunk 2]
```diff
@@ -19,6 +19,19 @@ namespace :cartodb do
       end
     end
 
+    desc "Purges broken table visualizations"
+    task :delete_inconsistent_tables => :environment do |t, args|
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply lib/tasks/viz_maintenance.rake]: Without user_table or related tables

File: lib/tasks/viz_maintenance.rake [hunk 3]
```diff
@@ -218,6 +231,10 @@ namespace :cartodb do
       (viz.table? && viz.related_tables.empty?) || (viz.derived? && viz.map.nil?)
     end
 
+    def inconsistent_table?(viz)
+      (viz.table? && viz.related_tables.empty?)
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply lib/tasks/viz_maintenance.rake]: I thought one thing is the user_table and the other one is the related tables

[the remark this answers is not in the document]
Comment (juanignaciosl) [reply lib/tasks/viz_maintenance.rake]: ^ If I  were right and it led to a single `viz.related_tables.empty?` check, I'd inline `inconsistent_table?`.

[the remark this answers is not in the document]
Comment (ethervoid) [reply lib/tasks/viz_maintenance.rake]: Changed take a look, that is what I wanted to do :)

File: lib/tasks/viz_maintenance.rake [hunk 4]
```diff
@@ -218,6 +233,10 @@ namespace :cartodb do
       (viz.table? && viz.related_tables.empty?) || (viz.derived? && viz.map.nil?)
     end
 
+    def inconsistent_table?(viz)
+      (viz.user_table? && viz.related_tables.empty?)
```
Comment (juanignaciosl) [inline lib/tasks/viz_maintenance.rake]: This fails:

```
irb(main):002:0> v.user_table
=> nil                                      
irb(main):003:0> v.user_table?
NoMethodError: undefined method `user_table?' for #<Carto::Visualization:0x00000006ddd078> 
```

Also tested with non-nil `user_table`. It fails because `user_table` is not an ActiveRecord attribute but a method.

Comment (ethervoid) [reply lib/tasks/viz_maintenance.rake]: 🤦‍♂️ I was thinking in `viz.user_table` without the question mark...changing

**Comment by ethervoid** [comment]
@juanignaciosl another :eye: ?

File: lib/tasks/viz_maintenance.rake [hunk 5]
```diff
@@ -19,6 +19,21 @@ namespace :cartodb do
...
     end
 
+    desc "Purges broken canonical visualizations without related tables"
+    task :delete_inconsistent_canonical_viz_without_tables => :environment do |_|
+      Carto::Visualization.joins("left join user_tables ut on visualizations.map_id = ut.map_id").where("visualizations.type = 'table' and ut.id is null").find_each do |viz|
+        puts "Checking for deletion --> User: #{viz.user.username} | Viz id: #{viz.id}"
+        if inconsistent_table?(viz)
+          puts "Deleting viz --> User: #{viz.user.username} | Viz id: #{viz.id}"
+          begin
+            viz.destroy
```
Comment (javitonino) [inline lib/tasks/viz_maintenance.rake]: You probably want `destroy!` here. `destroy` returns false on failure. `destroy!` raises an exception.

Comment (ethervoid) [reply lib/tasks/viz_maintenance.rake]: done! thx
